sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/estimator/ThroughputEstimatorTest.java

```diff
@@ -0,0 +1,172 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.spanner.changestreams.estimator;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.Timestamp;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.beam.repackaged.core.org.apache.commons.compress.utils.IOUtils;
+import org.apache.beam.repackaged.core.org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.beam.sdk.coders.Coder;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ThroughputEstimatorTest {
+  private static final double DELTA = 1e-10;
+  private static final int WINDOW_SIZE_SECONDS = 10;
+  private ThroughputEstimator<byte[]> estimator;
+
+  @Before
+  public void setup() {
+    final SizeEstimator<byte[]> sizeEstimator = new SizeEstimator<>(new TestCoder());
+    estimator = new ThroughputEstimator<>(WINDOW_SIZE_SECONDS, sizeEstimator);
+  }
+
+  @Test
+  public void testThroughputIsZeroWhenNothingsBeenRegistered() {
+    assertEquals(0D, estimator.get(), DELTA);
+    assertEquals(0D, estimator.getFrom(Timestamp.now()), DELTA);
+  }
+
+  @Test
+  public void testThroughputCalculation() {
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(2, 0), new byte[10]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(3, 0), new byte[20]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(5, 0), new byte[30]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(10, 0), new byte[40]);
+    assertEquals(10D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(11, 0)), DELTA);
+
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(20, 0), new byte[10]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(21, 0), new byte[20]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(21, 0), new byte[10]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(29, 0), new byte[40]);
+    assertEquals(8D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(30, 0)), DELTA);
+
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(31, 0), new byte[10]);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(35, 0), new byte[40]);
+    assertEquals(5D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(41, 0)), DELTA);
+
+    assertEquals(0D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(50, 0)), DELTA);
```

Comment:
```suggestion
    // No values in the past 10 seconds
    assertEquals(0D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(50, 0)), DELTA);
```

Reply:
Added (couldn't commit, because I changed the test class name)